app/src/main/java/com/github/houseorganizer/houseorganizer/shop/GroceriesActivity.java

```diff
@@ -0,0 +1,75 @@
+package com.github.houseorganizer.houseorganizer.shop;
+
+import android.annotation.SuppressLint;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.github.houseorganizer.houseorganizer.NavBar.NavBarHelpers;
+import com.github.houseorganizer.houseorganizer.R;
+import com.google.android.material.bottomnavigation.BottomNavigationView;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+public class GroceriesActivity extends AppCompatActivity {
+
+    private DocumentReference currentHouse;
+
+    private ShopListAdapter shopListAdapter;
+    private FirestoreShopList shopList;
+
+    @SuppressLint("NotifyDataSetChanged")
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_groceries);
+
+        currentHouse = FirebaseFirestore.getInstance().collection("households")
+                .document(getIntent().getStringExtra("house"));
+
+        initializeData();
+
+        findViewById(R.id.groceries_add).setOnClickListener(c -> {
+            if(shopListAdapter != null) shopListAdapter.addItem(this, shopList);
+            else initializeData();
+        });
+        findViewById(R.id.groceries_picked_up_button).setOnClickListener(c -> {
+            shopList.removePickedUpItems();
+            shopListAdapter.notifyDataSetChanged();
+        });
+
+
+        BottomNavigationView menu = findViewById(R.id.nav_bar);
+        menu.setSelectedItemId(R.id.nav_bar_cart);
+        menu.setOnItemSelectedListener(l -> {
+            Intent intent = NavBarHelpers.changeActivityIntent(l.getTitle().toString(),
+                    currentHouse, "Groceries", this);
+            if(intent==null)return false;
+            startActivity(intent);
+            return true;
+        });
+    }
+
+    private void initializeData(){
+        RecyclerView view = findViewById(R.id.groceries_recycler);
+        ShopListAdapter.initializeFirestoreShopList(currentHouse, FirebaseFirestore.getInstance())
+                .addOnCompleteListener(t -> {
+
+                    if(t.isSuccessful()){
+                        shopList = t.getResult().getFirestoreShopList();
+                        shopListAdapter = t.getResult();
+                        shopList.getOnlineReference().addSnapshotListener((d, e) -> {
+                            shopList = FirestoreShopList.buildShopList(d);
+                            shopListAdapter.setShopList(shopList);
+                        });
+                        view.setLayoutManager(new LinearLayoutManager(this));
+                        view.setAdapter(shopListAdapter);
+                    }else Log.e("Groceries", "Could not initialize list");
```

Comment:
using logAndToast might be nice here, so the user knows that there was a bug when it cannot be initialized

Reply:
Is logAndToast a function somewhere or should I just add a Toast after the log ?

Reply:
Alright it is added thanks ^^